AstroCloud CLI Reference Guide (#294)
astronomer/docs: cloud/cli-reference/astrocloud-dev-start.md

---
sidebar_label: "astrocloud dev start"
title: "astrocloud dev start"
id: astrocloud-dev-start
description: Reference documentation for astrocloud dev start.
---

> ## Description

Reviewer: Are we going with Overview or Description? I'm cool with either - let's just be consistent

Author: Definitely description, Overview is a typo

> Build your Astronomer project into a Docker image, spin up Airflow containers on your local machine, and run your project in a local Airflow environment.  

Reviewer: ```suggestion
Build your Astronomer project into a Docker image and spin up 3 Docker containers on your machine, each for a different Airflow component:

- **Postgres:** Airflow's metadata database
- **Webserver:** The Airflow component responsible for rendering the Airflow UI
- **Scheduler:** The Airflow component responsible for monitoring and triggering tasks

This command can be used regularly to start and run any Astronomer project locally. For more information, read [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).
```

Author: This is another part where I think it makes more sense to not have as much duplicative information in the reference doc. We should link out to the main doc and rely on that for explaining the Airflow components. Imo this reference should just serve as a quick, well, reference for someone who wants each option listed out in their browser or likes reading docs more than CLI help text.

---
sidebar_label: "astrocloud dev start"
title: "astrocloud dev start"
id: astrocloud-dev-start
description: Reference documentation for astrocloud dev start.
---

## Description

> Build your Astronomer project into a Docker image and spin up local a local Docker container for each Airflow component.

Reviewer: ```suggestion
Build your Astronomer project into a Docker image and restart all local containers for each Airflow component to run this Docker image.
```

Reviewer: Yea das fine - I replaced `restart` with just `start`

> This command can be used regularly to start and run any Astronomer project locally. For more information, read [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).

Reviewer: ```suggestion
This command can be used regularly to run an Astronomer project locally and apply changes as you test and troubleshoot DAGs. For more information, read [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).
```

Author: Not keen on mentioning troubleshooting if we're not linking out to troubleshoot docs, esp since the command itself isn't necessarily tied to troubleshooting in the same way that `astro dev kill` is
